plugins/rikki/heroeslounge/classes/discord/Attendance.php

```diff
@@ -68,8 +67,82 @@ public function CheckAttendance()
 
     public function IsOnServer($discordTag)
     {
-        $presentDiscordTags = $this->FetchUsers();
-        
-        return ($presentDiscordTags != false) ? in_array($discordTag, $presentDiscordTags) : false;
+        $presentUsers = $this->FetchUsers();
+        $userId = '';
+
+        if ($presentUsers) {
+          foreach ($presentUsers as $user) {
+            $userTag = $user["username"] . "#" . $user["discriminator"];
+            if ($userTag == $discordTag) {
+              $userId = $user["id"];
+              break;
+            }
+          }
+        }
+        return $userId
```

Comment:
The way this function is named it should return either true or false. Rename to GetUserId or GetDiscordUserId otherwise.